src/internal/clusterstate/v2.7.0_test.go

```diff
@@ -0,0 +1,67 @@
+package clusterstate
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	proto "github.com/gogo/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"github.com/jmoiron/sqlx"
+
+	v2_7_0 "github.com/pachyderm/pachyderm/v2/src/internal/clusterstate/v2.7.0"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dockertestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testetcd"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+// Create sample test data in collections.projects table
+func setupTestData(t *testing.T, ctx context.Context, db *sqlx.DB) {
+	t.Helper()
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err)
+	defer tx.Rollback()
+	for i := 0; i < 12; i++ {
```

Comment:
This feels a little too coupled to implementation details.  It would be best to go through the API server and collections API to add "old" records.  Otherwise the test is just testing how you think the API server works, not how it actually works, which isn't ideal.

Reply:
Valid point, and I'm a bit torn on this. I wanted to decouple my test code from collections package as much as possible, because eventually we will be removing it. However, going through the API server makes sense!

Reply:
As I read down a little farther in the PR, I decided I don't object to this existing, but the scenario I'm talking about does need to be covered.

It's going to be more complicated to have tests that start by running the old version, but we're kind of changing the wings of an airplane while it's flying, so it's going to be more painful than starting from scratch.

Reply:
Hmm after thinking about it a bit more, I'm not sure how to achieve your scenario 😓.
The big issue is that we already changed the implementation of our API server to use the new tables, and I'm not sure how to swap this implementation out for the old collections style.

Reply:
Start up pachyderm 2.6, make the API calls, upgrade to 2.7, check the data.  The old API server is running at the start of the test.  If the client changes so significantly that this isn't possible, then we need to put the pachctl binary from 2.6 into the CI environment and use that to create the objects in 2.6.